### ISSUE TITLE
smp=1 leads to a node failure after startup

### ISSUE THREAD

**Issue by kostja** [scylladb/scylla#4492]
I have a version of Scylla built from source, the master branch as of today. 
I am using a simple configuration file (attached) and start in developer_mode=true.
I provide smp=1 on the command line.
The instance crashes after some a few second of uptime:
```
INFO  2019-05-13 20:43:17,356 [shard 0] init - Signal received; shutting down
INFO  2019-05-13 20:43:17,356 [shard 0] compaction_manager - Asked to stop
INFO  2019-05-13 20:43:17,357 [shard 0] compaction_manager - Stopped
INFO  2019-05-13 20:43:17,357 [shard 0] view - Stopping view builder
terminate called after throwing an instance of 'seastar::no_sharded_instance_exception'
  what():  sharded instance does not exists
```

**Comment by avikivity** [comment]
`INFO  2019-05-13 20:43:17,356 [shard 0] init - Signal received; shutting down` - you send SIGINT (or SIGTERM) to the instance, and it crashed as it was going down. Doesn't reproduce for me though,

**Comment by kostja** [comment]
I think the key is that I had an empty data dir.

**Comment by kostja** [comment]
[db.tar.gz](https://github.com/scylladb/scylla/files/3174603/db.tar.gz)

**Comment by kostja** [comment]
Trace:
```
Thread 1 "scylla" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff5143535 in __GI_abort () at abort.c:79
#2  0x00007ffff5e74642 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff5e80186 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff5e7f1e9 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff5e7fb84 in __gxx_personality_v0 ()
   from /lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff53197a3 in ?? () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#7  0x00007ffff531a006 in _Unwind_Resume ()
   from /lib/x86_64-linux-gnu/libgcc_s.so.1
#8  0x000055558b58a32c in seastar::future_state<>::~future_state() (
    this=0x7fffe912a528, __in_chrg=<optimized out>)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/core/future.hh:311
#9  seastar::future<>::~future() (this=<optimized out>, 
    __in_chrg=<optimized out>)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/core/future.hh:841
#10 <lambda()>::<lambda()>::<lambda()>::operator()(void) const (
    __closure=0x7fffe912c110) at main.cc:920
#11 0x000055558b5f9b4a in seastar::deferred_action<main(int, char**)::<lambda()>::<lambda()>::<lambda()> >::~deferred_action(void) (this=0x7fffe912c110, 
    __in_chrg=<optimized out>)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/util/defer.hh:44
#12 0x000055558b5d41ec in <lambda()>::<lambda()>::operator()(void) const (
    __closure=0x612000002ef0) at main.cc:921
#13 0x000055558b626bfe in seastar::apply_helper<main(int, char**)::<lambda()>::<lambda()>, std::tuple<>&&, std::integer_sequence<long unsigned int> >::apply(<lambda()>::<lambda()> &&, std::tuple<> &&) (func=..., args=...)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/core/apply.hh:35
#14 0x000055558b626c8a in seastar::apply<main(int, char**)::<lambda()>::<lambda()> >(<lambda()>::<lambda()> &&, std::tuple<> &&) (func=..., args=...)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/core/apply.hh:43
#15 0x000055558b626dd5 in seastar::futurize<int>::apply<main(int, char**)::<lambda()>::<lambda()> >(<lambda()>::<lambda()> &&, std::tuple<> &&) (func=..., 
    args=...)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/core/future.hh:1338--Type <RET> for more, q to quit, c to continue without paging--

#16 0x000055558b608a33 in seastar::<lambda(seastar::async(seastar::thread_attributes, Func&&, Args&& ...) [with Func = main(int, char**)::<lambda()>::<lambda()>; Args = {}; seastar::futurize_t<typename std::result_of<typename std::decay<_Tp>::type(std::decay_t<_Types>...)>::type> = seastar::future<int>]::work&)>::<lambda()>::operator()(void) const (this=0x61a0000090a8)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/core/thread.hh:324
#17 0x000055558b656c4c in seastar::noncopyable_function<void()>::direct_vtable_for<seastar::async(seastar::thread_attributes, Func&&, Args&& ...) [with Func = main(int, char**)::<lambda()>::<lambda()>; Args = {}; seastar::futurize_t<typename std::result_of<typename std::decay<_Tp>::type(std::decay_t<_Types>...)>::type> = seastar::future<int>]::<lambda(seastar::async(seastar::thread_attributes, Func&&, Args&& ...) [with Func = main(int, char**)::<lambda()>::<lambda()>; Args = {}; seastar::futurize_t<typename std::result_of<typename std::decay<_Tp>::type(std::decay_t<_Types>...)>::type> = seastar::future<int>]::work&)> mutable::<lambda()> >::call(const seastar::noncopyable_function<void()> *) (
    func=0x61a0000090a0)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/util/noncopyable_function.hh:71
#18 0x000055558cb46050 in seastar::noncopyable_function<void ()>::operator()() const (this=0x61a0000090a0)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/util/noncopyable_function.hh:145
#19 0x000055559651d9f4 in seastar::thread_context::main (this=<unavailable>)
    at ../../src/core/thread.cc:317
```

**Comment by kostja** [comment]
Just in case the other thread data is relevant:
```
(gdb) info thread
  Id   Target Id                                     Frame 
* 1    Thread 0x7ffff46588c0 (LWP 13649) "scylla"    0x000055558b58a32c in seastar::future_state<>::~future_state() (this=0x7fffe912a528, 
    __in_chrg=<optimized out>)
    at /opt/local/work/scylla/scylla/seastar/include/seastar/core/future.hh:311
  2    Thread 0x7fffe93ff700 (LWP 13651) "syscall-0" __libc_read (nbytes=8, 
    buf=0x7fffe93d05d0, fd=6) at ../sysdeps/unix/sysv/linux/read.c:26
(gdb) thread 2
[Switching to thread 2 (Thread 0x7fffe93ff700 (LWP 13651))]
#0  __libc_read (nbytes=8, buf=0x7fffe93d05d0, fd=6)
    at ../sysdeps/unix/sysv/linux/read.c:26
26	../sysdeps/unix/sysv/linux/read.c: No such file or directory.
(gdb) bt
#0  __libc_read (nbytes=8, buf=0x7fffe93d05d0, fd=6)
    at ../sysdeps/unix/sysv/linux/read.c:26
#1  __libc_read (fd=6, buf=0x7fffe93d05d0, nbytes=8)
    at ../sysdeps/unix/sysv/linux/read.c:24
#2  0x00007ffff75d8860 in ?? () from /lib/x86_64-linux-gnu/libasan.so.5
#3  0x0000555595f77049 in seastar::thread_pool::work (this=0x61d000003280, 
    name=...) at ../../src/core/reactor.cc:4764
#4  0x0000555595f76571 in seastar::thread_pool::<lambda()>::operator()(void) const (__closure=0x60300003ea70) at ../../src/core/reactor.cc:4752
#5  0x00005555960180ae in std::_Function_handler<void(), seastar::thread_pool::thread_pool(seastar::reactor*, seastar::sstring)::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...) at /usr/include/c++/8/bits/std_function.h:297
#6  0x000055558e351045 in std::function<void ()>::operator()() const (
    this=0x60300003eaa0) at /usr/include/c++/8/bits/std_function.h:687
#7  0x0000555596eafcff in seastar::posix_thread::start_routine (
    arg=0x60300003eaa0) at ../../src/core/posix.cc:52
#8  0x00007ffff6587182 in start_thread (arg=<optimized out>)
    at pthread_create.c:486
#9  0x00007ffff523bb1f in clone ()
    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by slivne** [comment]
@kostja can you please provide instruction on how to reproduce

**Comment by kostja** [comment]
Start the binary  and hit ctrl-c at random times, it fails every once in a while.

**Comment by bhalevy** [comment]
This looks exactly like #4589 that was fixed by efa7951ea5733a9f064de83dc0eacbec9d958791
